src/Workspaces/Core/Portable/LinkedFileDiffMerging/LinkedFileDiffMergingSession.cs

```diff
@@ -19,88 +17,112 @@ namespace Microsoft.CodeAnalysis;
+            var filePath = newDocument.FilePath;
+            Contract.ThrowIfNull(filePath);
 
-        var linkedFileMergeResults = new List<LinkedFileMergeResult>();
+            if (!filePathToNewDocumentsAndHashes.TryGetValue(filePath, out var newDocumentsAndHashes))
+            {
+                newDocumentsAndHashes = [];
+                filePathToNewDocumentsAndHashes.Add(filePath, newDocumentsAndHashes);
+            }
+
+            var newText = await newDocument.GetValueTextAsync(cancellationToken).ConfigureAwait(false);
+            var newContentHash = newText.GetContentHash();
+            // Ignore any linked documents that we have the same contents as.  
+            if (newDocumentsAndHashes.Any(t => t.newContentHash.SequenceEqual(newContentHash)))
+                continue;
+
+            newDocumentsAndHashes.Add((newDocument, newContentHash));
+        }
 
         var updatedSolution = newSolution;
-        foreach (var linkedDocumentsWithChanges in linkedDocumentGroupsWithChanges)
+        var linkedFileMergeResults = new List<LinkedFileMergeResult>();
+
+        foreach (var (filePath, newDocumentsAndHashes) in filePathToNewDocumentsAndHashes)
         {
-            var documentInNewSolution = newSolution.GetDocument(linkedDocumentsWithChanges.First());
+            Contract.ThrowIfTrue(newDocumentsAndHashes.Count == 0);
 
-            // Ensure the first document in the group is the first in the list of 
-            var allLinkedDocuments = documentInNewSolution.GetLinkedDocumentIds().Add(documentInNewSolution.Id);
-            if (allLinkedDocuments.Length == 1)
-            {
+            // Don't need to do anything if this document has no linked siblings.
+            var firstNewDocument = newDocumentsAndHashes[0].newDocument;
+            var relatedDocuments = newSolution.GetRelatedDocumentIds(firstNewDocument.Id);
+            if (relatedDocuments.Length == 1)
                 continue;
-            }
 
-            SourceText mergedText;
-            if (linkedDocumentsWithChanges.Count() > 1)
+            if (newDocumentsAndHashes.Count == 1)
             {
-                var mergeGroupResult = await MergeLinkedDocumentGroupAsync(allLinkedDocuments, linkedDocumentsWithChanges, sessionInfo, mergeConflictHandler, cancellationToken).ConfigureAwait(false);
-                linkedFileMergeResults.Add(mergeGroupResult);
-                mergedText = mergeGroupResult.MergedSourceText;
+                // The file has linked siblings, but we collapsed down to only one actual document change.  Ensure that
+                // any linked files have that same content as well.
+                var firstSourceText = await firstNewDocument.GetValueTextAsync(cancellationToken).ConfigureAwait(false);
+                updatedSolution = updatedSolution.WithDocumentTexts(
+                    relatedDocuments.SelectAsArray(d => (d, firstSourceText)));
```

Comment:
this is the next meat part. if we've only truly come up with one change to a doc, then we don't even have to do any diff/merge work. we can just take that change and apply to all linked files.